### PR TITLE
Adding Sorting

### PR DESCRIPTION
**Pull request by iluxav**

### CODE REVIEW

File: src/DataTable/DataTable.spec.js [hunk 1]
```diff
@@ -121,21 +121,75 @@ describe('Table', () => {
     });
   });
 
+
+  describe('Sortable column titles', () => {
+    const tests = [
```
Comment (lbelinsk) [inline src/DataTable/DataTable.spec.js]: You are not using tests variable..

File: src/DataTable/DataTable.spec.js [hunk 2]
```diff
@@ -121,21 +121,75 @@ describe('Table', () => {
...
+      { handler: 'onRowClick', driverMethod: 'clickRow' },
+      { handler: 'onMouseEnterRow', driverMethod: 'mouseEnterRow' },
+      { handler: 'onMouseLeaveRow', driverMethod: 'mouseLeaveRow' }
+    ];
+
+    it('should display sortable title', () => {
+      const props = {
```
Comment (lbelinsk) [inline src/DataTable/DataTable.spec.js]: you are defining const props 3 times in each it block. Better to define it in the describe block once

File: src/DataTable/DataTable.spec.js [hunk 3]
```diff
@@ -121,21 +121,75 @@ describe('Table', () => {
...
+          { title: 'B', render: row => row.b }
+        ],
+        onSortClick: jest.fn()
+      };
+      const driver = createDriver(<DataTable {...props} />);
+      driver.clickSort(2);
+      expect(props.onSortClick).not.toHaveBeenCalled();
+    });
+  });
+
```
Comment (lbelinsk) [inline src/DataTable/DataTable.spec.js]: I am missing a test to check that the icon is getting flipped onClick. wdyt?

File: stories/DataTable/ExampleSortable.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    return this.state.data.sort((a, b) => desc ? ~~(field ? a[field] < b[field] : a < b) : ~~(field ? a[field] > b[field] : a > b));
+  }
+
+  render() {
+    return (
+      <div style={style}>
+        <DataTable
+          dataHook='story-data-table'
+          data={this.state.data}
+          onRowClick={(row, rowNum) => {
```
Comment (lbelinsk) [inline stories/DataTable/ExampleSortable.js]: empty callback. can remove it

File: stories/DataTable/ExampleSortable.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          }}
+          onSortClick={(col, colNum) => this.handleSortClick(colNum)}
+          infiniteScroll
```
Comment (lbelinsk) [inline stories/DataTable/ExampleSortable.js]: please remove InfinitScrool and ItemsPerPage, not relevant and can confuse the users

File: stories/DataTable/ExampleSortable.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    const filelds = {
+      '1': 'firstName',
+      '2': 'lastName'
+    };
+    const sortedData = this.sortDataByField(filelds[colNum], desc);
+    this.setState({ sort, data: sortedData });
+  }
+
+  sortDataByField(field, desc) {
+    return this.state.data.sort((a, b) => desc ? ~~(field ? a[field] < b[field] : a < b) : ~~(field ? a[field] > b[field] : a > b));
```
Comment (lbelinsk) [inline stories/DataTable/ExampleSortable.js]: I am not familiar with this ~~ syntax :)
What is it doing?

File: src/DataTable/DataTable.js [hunk 7]
```diff
@@ -4,6 +4,7 @@ import s from './DataTable.scss';
 import classNames from 'classnames';
 import InfiniteScroll from './InfiniteScroll';
 import WixComponent from '../BaseComponents/WixComponent';
+import ArrowVertical from '../Icons/dist/components/ArrowVertical';
```
Comment (lbelinsk) [inline src/DataTable/DataTable.js]: I would think the ArrowDownThin will be more suitable here.
You should consult with your UX, but most importantly with the new UX of wix-style-react Ben Benhorin.

File: src/DataTable/DataTable.js [hunk 8]
```diff
@@ -132,32 +133,46 @@ class DataTable extends WixComponent {
...
       height: this.props.thHeight,
-      fontSize: this.props.thFontSize
+      fontSize: this.props.thFontSize,
+      cursor: column.sortable === undefined ? 'arrow' : 'pointer'
     };
-    return <th style={style} key={colNum}>{column.title}</th>;
+
+    let optionalHeaderCellProps = {};
+    if (column.sortable) {
+      optionalHeaderCellProps.onClick = () => this.props.onSortClick(column, colNum);
```
Comment (lbelinsk) [inline src/DataTable/DataTable.js]: the onSortClick prop is not "isRequired", doesn't have default noop prop. You can add one of them or wrap this usage with an if condition

File: src/DataTable/DataTable.js [hunk 9]
```diff
@@ -132,32 +133,46 @@ class DataTable extends WixComponent {
...
-    const classes = classNames({[s.important]: column.important});
+    const classes = classNames({ [s.important]: column.important });
     return <td className={classes} key={colNum}>{column.render && column.render(rowData, rowNum)}</td>;
   };
 
+  renderSortingArrow = (sortDescending, colNum) => {
+    if (sortDescending === undefined) {
+      return null;
+    }
+    const sortDirectionClassName = sortDescending === true ? s.sortArrowAsc : s.sortArrowDesc;
```
Comment (lbelinsk) [inline src/DataTable/DataTable.js]: === true is not really needed, up to you..

File: src/DataTable/DataTable.spec.js (hunk 3 above)
Comment (iluxav) [inline src/DataTable/DataTable.spec.js]: Icon position reflects the true/false prop that passed from outside. So I'm not sure if I really need to test it css, not sure...

**Pull request merged**